src/router/chainable_router.rs

```diff
@@ -0,0 +1,71 @@
+use hyper::method::Method;
+use middleware::Middleware;
+use nickel::Nickel;
+use router::http_router::{HttpRouter};
+
+pub struct ChainableRoute<'a, D: Sync + Send + 'static = ()> {
+    path: String,
+    nickel: &'a mut Nickel<D>
```

Comment:
Wondering if this should be more generic so it can handle:

```rust
let mut x = Router::new();
x.route(..)
 .get(..)
 .post(..);
```

I think if people want to structure their app they probably want to deal with subrouters (which they can mount at certain endpoints) rather than passing around `Nickel` instances.